Topics/Tech_Stacks/React_Prop_Drilling.md

```diff
@@ -0,0 +1,113 @@
+# React Bad Practices - Prop Drilling
+
+# Table of Contents
+1. [What is Props Drilling?](#what-is-props-drilling)
+2. [Why is Props Drilling Bad?](#why-is-props-drilling-bad)
+3. [Example of Props Drilling](#example-of-props-drilling)
+4. [Avoiding Props Drilling](#avoiding-props-drilling)
+5. [Alternative Solutions](#alternative-solutions)
+6. [Final Remark](#final-remark)
+7. [Sources](#sources)
+
+## What is Props Drilling?
+In React.js, props drilling refers to the process of passing props down through multiple layers of components. It happens when a component needs to pass data to a deeply nested child component, and each intermediate parent component in the hierarchy needs to receive and pass along the props. This is not a huge issue in small lightweight applications but can become a large issue when the application scales.
+
+## Why is Props Drilling Bad?
+Props drilling can lead to several issues in a React application:
+
+- **Maintenance:** As the application grows, managing and updating props through multiple layers becomes extremely cumbersome and error-prone. It makes the codebase harder to understand and maintain.
+
+- **Code Smells:** Props drilling often indicates a design flaw in the application's component hierarchy. It violates the principles of encapsulation and modularity, making components less reusable and more tightly coupled.
+
+- **Reduced Component Isolation:** Components become overly dependent on their parent components, breaking the idea of encapsulation. Changes in one layer may require modifications in several other layers. This problem is evident when the application is large.
```

Comment:
The explanation is clear. You might also want to mention the potential impact on performance due to unnecessary re-rendering of components due to the chain of prop changes.